### PR TITLE
 BAU: report coverage to Codacy

### PR DESCRIPTION
**Pull request by bjgill**
Aggregate the reports from each subproject. Added github authentication so we don't fall foul of the API rate limiting.

See https://trello.com/c/qPUHgpLn

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,27 @@
 language: java
 env:
-  - VERIFY_USE_PUBLIC_BINARIES=true
+  global:
+    - VERIFY_USE_PUBLIC_BINARIES=true
+    # CODACY_PROJECT_TOKEN
+    - secure: "oxKr68jSBUMtqG9djhRpkt1Wek7WNqbs5EwUXcxLAx4izMbQR2MpXS7wj7IF4FXCq2ZyHIuPookzi4aqZyA9LEcEKYIotr8fq2QZ27EXrNsAXNwWJNnYO1GfP7PuDwcNKaFrubxCshA6Yga8nOV/11A1hq8ngiQ6YRMSEvU64NXqvEgp4jVIDRAW6ZHtEWn9BiNiLm1Bf/5wQ7+VLWIzSPEjdQc1qIwlpcr1c+IzmWkcAoOSRlZQlB1CM5ZAo3OY+9lACr2vhL07AufkwiiCQ0uMBv/zcvG0iexp8U9IA1EMNgvW+zsEyVYyVf3AvQWsAOV+RQn2Pia3tfHdi9miWIzIjvwVnVCK3eWkBnGoWm/k7V30aeV8yRsMwTzaSgFMLXk2C6LJfIGbH7RZE96K/SkJ4wYMJe0PbZwoYaE6wX44AVrDZpSFPeop7rdbfRBBmzHTs0npRvN7bPp+PGbQ3jOK/PMmkrlm7z7OX55DHglZfoywyQj3+3k42bbjXe75eIPr6KQ+DA0pTGUfSoJRnjhIck9tGQ32EFGewwn+eDkv/iRchyzvZF2vmJe0TAtRCQR5LSGj/TlvFdvPHMGg+3n5qzbHMpIS1WN/SEKp++6oxMZbgvt/of8Tvlv3OQb43gFtDFbvp6Jyz6WAH4fm24oehw+4Cq95TEsqBSJPCa8="
```
Comment (JHjava) [inline .travis.yml]: For info, what does this token and the one below get used for?

Comment (bjgill) [reply .travis.yml]: `CODACY_PROJECT_TOKEN` is needed by `codacy-coverage-reporter-assembly.jar` to upload the coverage to Codacy. It is the API token for the verify-smal-libs project in Codacy.

`GITHUB_PAT` is needed for github authentication (of the curl requests) so we don't fall foul of the API rate limiting.

Comment (bjgill) [reply .travis.yml]: Do you think it'd be worth adding more information to those comments, then?

Comment (JHjava) [reply .travis.yml]: Could be useful. Naturally when I see a secure tag and a token checked in to a public repo, it worries me slightly. I'm assuming it's fine to have though but I think it just the naming of the tag which we don't define.

**Pull request merged**